core/2d/CCMotionStreak.cpp

```diff
@@ -101,7 +101,12 @@ bool MotionStreak::initWithFade(float fade, float minSeg, float stroke, const Co
     _stroke    = stroke;
     _fadeDelta = 1.0f / fade;
 
-    double fps = 1 / _director->getAnimationInterval();
+    // Fix #629 Motion streak vsync off crash
+    double interval = _director->getAnimationInterval();
+    double fps      = 1 / interval;
+    if (roundf(fps) == INFINITY)
```

Comment:
```suggestion
    if (roundf(fps) > 240.0)
```

Reply:
that might be a good idea actually, making 240 the soft limit for the streak